lib/react.dart

```diff
@@ -5,14 +5,28 @@
 /// A Dart library for building UI using ReactJS.
 library react;
 
+import 'dart:html' show Element;
+
 /// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
 /// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
 abstract class Component {
+  Map _props;
+  Map _state = {};
+  dynamic _ref;
+
   /// ReactJS `Component` props.
-  Map props;
+  Map get props => _props;
+  set props(Map value) => _props = value;
```

Comment:
As opposed to making these getters/setters, can we just use the `@virtual` annotation instead? (See [meta package](https://pub.dartlang.org/packages/meta) 1.0.4 changelog)